### ISSUE TITLE
Any idea to add Conventional Commits support

### ISSUE THREAD

**Issue by linux-china** [AmrDeveloper/GQL#109]
Most developers adopt [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) for commit, such as 

```
feat: allow provided config object to extend other configs
feat(lang): add Polish language
docs: correct spelling of CHANGELOG
```

It's ok to add `type` column for commits table?   If not Conventional Commits style, and type is null.

With type column introduce, and it's easy to filter conventional commits.

```
gitql -q 'select * from commits where type = "feat"'
```

**Comment by AmrDeveloper** [comment]
Hello @linux-china,

I like the idea but i am thinking to provide the features as a function to and not add new table for each style in the schema for example

'SELECT * FROM commits WHERE commit_conventional(message) = "feat"'

This function can be implemented only on gitql and not in the gitql sdk

**Comment by AmrDeveloper** [comment]
@linux-china Done, will be available on next release

![commit_conventional](https://github.com/AmrDeveloper/GQL/assets/23631699/83016e10-cf5a-4d10-9075-9179532802b1)

Thank you,
Amr Hesham

**Comment by linux-china** [comment]
Fixed by 0.24. Thanks :)
